iotilecore/iotile/core/hw/hwmanager.py

```diff
@@ -785,7 +788,47 @@ def get_proxy(self, short_name):
         if short_name not in self._name_map:
             return None
 
-        return self._name_map[short_name][0]
+        proxy_match = self.find_correct_proxy_version(self._name_map[short_name], version)
+        print("proxy_match: ", proxy_match)
+        return proxy_match if proxy_match is not None else self._name_map[short_name][0] 
+
+    def find_correct_proxy_version(self, proxies, version):
+        """Retrieves the ModuleVersion of each proxy and match it with the tile version"""
+
+        proxy_details = {}
+        tile_version = SemanticVersion(version[0], version[1], version[2])
+        latest_version = SemanticVersion(0, 0, 0)
+
+        for proxy in proxies:
+            proxy_details[proxy] = {}
+            try:
+                # If proxy has ModuleVersion(), get the SemanticVersionRange
+                module_version = proxy.ModuleVersion()
+                proxy_details[proxy]['ModuleVersion'] = module_version
+            except AttributeError:
+                # If proxy does not have ModuleVersion(), use the name of the support package
+                stripped_proxy_version = str(proxy).split('.')[0].split('_')[-1]
+                module_version = SemanticVersionRange.FromString('^' + stripped_proxy_version + '.0.0')
+                proxy_details[proxy]['ModuleVersion'] = module_version
+
+            # Keep track of each proxy's minimum version required
+            proxy_details[proxy]['MinVersion'] = module_version._disjuncts[0][0][0]
+
+            if module_version.check(tile_version):
```

Comment:
Make sure `proxy.ModuleVersion()` returns a `SemanticVersionRange`, otherwise I'm not sure this check will work.

Reply:
Sure, I can check that easily with a `isinstance(ModuleVersion, SemanticVersionRange)`. Curious, what would I do if it wasn't? Crap out and raise an error? Technically it would only happen in the `try` block, maybe it could just redirect to the except block i have for `AttributeError`.